PROTO_tests/tests/numeric_test.py

```diff
@@ -206,13 +205,242 @@ def test_sin(self, num_type):
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.arcsin(na), ak.arcsin(pda).to_ndarray())
+        with pytest.raises(TypeError):
+            ak.arcsin(np.array([range(0, 10)]).astype(num_type))
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_arccos(self, num_type):
+        print("mytest")
+        if num_type == ak.uint64:
+            na = np.arange(0, 2).astype(num_type)
+        elif num_type == ak.float64:
+            na = np.linspace(-1, 1)
+        else:
+            na = np.arange(-1, 2).astype(num_type)
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.arccos(na), ak.arccos(pda).to_ndarray())
+        with pytest.raises(TypeError):
+            ak.arccos(np.array([range(0, 10)]).astype(num_type))
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_arctan(self, num_type):
+        if num_type == ak.float64:
+            na = np.linspace(0, 10)
+        else:
+            na = np.arange(0, 10).astype(num_type)
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray())
+        with pytest.raises(TypeError):
+            ak.arctan(np.array([range(0, 10)]).astype(num_type))
+
+        # Edge case: infinities
+        if num_type == ak.float64:
+            na = np.array([np.inf, -np.inf])
+            pda = ak.array(na)
+            assert np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray(), equal_nan=True)
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    @pytest.mark.parametrize("denom_type", NO_BOOL)
+    def test_arctan2(self, num_type, denom_type):
+        if num_type == ak.float64:
+            na_num = np.linspace(0, 1, 10)
+            pda_num = ak.array(na_num, dtype=num_type)
+        else:
+            na_num = np.arange(10, 0, -1).astype(num_type)
+            pda_num = ak.array(na_num, dtype=num_type)
+        if denom_type == ak.float64:
+            na_denom = np.linspace(0, 10, 10)
+            pda_denom = ak.array(na_denom, dtype=denom_type)
+        else:
+            na_denom = np.arange(0, 10).astype(denom_type)
+            pda_denom = ak.array(na_denom, dtype=denom_type)
+
+        num = np.array([1]).astype(num_type)
+        denom = np.array([5]).astype(denom_type)
+
+        assert np.allclose(np.arctan2(na_num, na_denom), ak.arctan2(pda_num, pda_denom).to_ndarray())
+        assert np.allclose(np.arctan2(num[0], na_denom), ak.arctan2(num[0], pda_denom).to_ndarray())
+        assert np.allclose(np.arctan2(na_num, denom[0]), ak.arctan2(pda_num, denom[0]).to_ndarray())
+
+        with pytest.raises(TypeError):
+            ak.arctan2(
+                np.array([range(0, 10)]).astype(num_type), np.array([range(10, 20)]).astype(num_type))
+            ak.arctan2(num[0], np.array([range(10, 20)]).astype(num_type))
+            ak.arctan2(np.array([range(0, 10)]).astype(num_type), denom[0])
+
+        # Edge case: Infinities and Zeros
```

Comment:
we say `Infinities and Zeros` but i don't think we're testing any zeros

Reply:
Updated, the zeros are tested in the main test block